### PR TITLE
Fix overflowing UI & buttons in Signature Request screens 

### LINKED ISSUE #18224: [Bug]: Opensea security provider - buttons and text displayed outside the modal on notification page
### Describe the bug

When the opensea security provider is enabled, as there's warning message displayed in notification window, the text and buttons displayed outside of the modal when user try to perform **Sign Typed Data** or **SIWE with malformed message**.

https://user-images.githubusercontent.com/105063779/225944150-1b69300f-d7b1-491e-8288-9dbc05de5e0f.mov

### Steps to reproduce

1. Enable Opensea security provider checker in extension settings experimental tab
2. Go to [test-dapp](https://metamask.github.io/test-dapp/)
3. Click **Sign in with ethereum (malformed)**
4. Check reject and sign button are displayed outside the modal on extension notification
5. Click **Sign Typed Data** on testdapp
6. Check text and buttons are displayed outside the modal on extension notification

Expected:
Text and buttons should be displayed inside the modal even the warning message is displayed. A scrollbar could be added as the following:

Screenshot 2023-03-17 at 16 09 59

### Error messages or log output

_No response_

### Version

10.28.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

Fixes https://github.com/MetaMask/metamask-extension/issues/18224

When the header has too much content, it may push down the rows of data to the point it is not visible, and it may push the footer so it is overflowing.

To fix this, we will ~add a second scrollbar and~ fix the flex styles * and move the scroll bar in this PR part 2 - https://github.com/MetaMask/metamask-extension/pull/18305. 

Additionally: 
- Fix unnecessary spacing between rows when there is too much room
- Fix uneven buttons with no padding in between when window is x-small

### Before
Screen Shot 2023-03-20 at 4 03 28 PM

Screen Shot 2023-03-20 at 5 05 56 PM
Screen Shot 2023-03-20 at 4 57 40 PM
Screen Shot 2023-03-20 at 5 20 22 PM

### After

Screen Shot 2023-03-20 at 5 34 45 PM
Screen Shot 2023-03-20 at 5 39 25 PM

Screen Shot 2023-03-20 at 4 08 34 PM

Screen Shot 2023-03-20 at 5 20 35 PM

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/signature-request-original/index.scss [hunk 1]
```diff
@@ -62,11 +62,11 @@
   }
 
   &__body {
-    width: 100%;
```
Comment (digiwand) [inline ui/components/app/signature-request-original/index.scss:65]: no need for width 100% since this would be inherited as a `div` element

**Review by jpuri — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

File: ui/components/ui/page-container/index.scss [hunk 2]
```diff
@@ -190,7 +190,7 @@
       margin-right: 0;
 
       &:first-of-type {
-        margin-bottom: 0;
+        margin-right: 4px;
```
Comment (georgewrmarshall) [inline ui/components/ui/page-container/index.scss:193]: What is this for?

**Review by georgewrmarshall — APPROVED** [review]
LGTM! with one question

**Comment by digiwand** [comment]
Thanks @georgewrmarshall! [question answered above](https://github.com/MetaMask/metamask-extension/pull/18247#discussion_r1146449040)

**Comment by digiwand** [comment]
Apologies @danjm @jpuri @brad-decker @georgewrmarshall. I forgot to add the scroll code. Somehow it got left behind on my local machine

additional PR created here https://github.com/MetaMask/metamask-extension/pull/18305

**Pull request merged**